subscription/views.py

```diff
@@ -999,41 +1000,73 @@ def post(self, request):
+#     except PremiumUsageUnits.DoesNotExist:
+#         # If no exact match, try regex matching
+#         for premium_page in PremiumUsageUnits.objects.all():
+#             if re.match(premium_page.url, url):
+#                 return premium_page.units_consumed
+#     return 1  # Default to 1 unit if not found in PremiumUsageUnits
+
+def get_premium_units(url):
+    logger.debug("get_premium_units " + url)
+    try:
+        # First, try an exact match
+        premium_page = PremiumUsageUnits.objects.get(url=url)
+        return premium_page.units_consumed
+    except PremiumUsageUnits.DoesNotExist:
+        # If no exact match, try containment check
+        for premium_page in PremiumUsageUnits.objects.all():
+            if premium_page.url in url:
+                return premium_page.units_consumed
+        
+        # If still no match, try regex matching as a fallback
+        for premium_page in PremiumUsageUnits.objects.all():
+            if re.match(premium_page.url, url):
+                return premium_page.units_consumed
+    
+    return 1  # Default to 1 unit if not found in PremiumUsageUnits
 
 class CheckLimit(APIView):
     requireApiKey = False
 
     def get(self, request):
-        # timestamp = request.GET.get('timestamp')
-        # complete_uri = request.GET.get('complete_uri')
+        complete_uri = request.GET.get('uri')
         party_id = request.GET.get('party_id')
-        # logger.debug("CheckLimit view called")
-
-        if not all([party_id]):
-            return Response({"error": "Missing party_id parameter"}, status=status.HTTP_400_BAD_REQUEST)
 
-        # logger.debug("all variables present")
-        # Check if the requested URI is a paid page
-        # if not is_paid_page(complete_uri):
-        #     return Response({"status": "ok", "message": "Free page access granted"})
-
-        # logger.debug("paid page")
+        if not all([party_id, complete_uri]):
+            return Response({"error": "Missing required parameters"}, status=status.HTTP_400_BAD_REQUEST)
 
+        units_required = get_premium_units(complete_uri)
+        
+        try:
+            limit_value_object = LimitValue.objects.get(limitId=1)
+            warningLimit = limit_value_object.val
+        except LimitValue.DoesNotExist:
+            return Response({"error": "Warning limit not found in the database"}, status=status.HTTP_500_INTERNAL_SERVER_ERROR)
+        
         try:
-            user_bucket = get_object_or_404(UserBucketUsage, partyId_id=party_id)
-            warningLimit = 5
-            if user_bucket.remaining_units > 0 and user_bucket.expiry_date > timezone.now():
+            user_bucket = UserBucketUsage.objects.get(partyId_id=party_id)
+            if user_bucket.remaining_units >= units_required and user_bucket.expiry_date > timezone.now():
                 if user_bucket.remaining_units == warningLimit:
                     return Response({"status": "Warning"})
```

Comment:
Please convert those response status codes into constant variables like https://github.com/tair/proxy/blob/83dd26ceab7659f6fc3ea0bf0b9ef2a25e5a6f5f/proxy/src/org/phoenixbioinformatics/proxy/Proxy.java#L145 does. Also check if other spots have the same issue.